src/neptune/objects/with_backend.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2024, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+__all__ = ["WithBackend"]
+
+import abc
+from contextlib import AbstractContextManager
+from typing import Optional
+
+from neptune.internal.backends.api_model import Project
+from neptune.internal.backends.factory import get_backend
+from neptune.internal.backends.neptune_backend import NeptuneBackend
+from neptune.internal.backends.project_name_lookup import project_name_lookup
+from neptune.internal.container_type import ContainerType
+from neptune.internal.id_formats import (
+    QualifiedName,
+    UniqueId,
+    conform_optional,
+)
+from neptune.internal.utils import verify_type
+from neptune.objects.mode import Mode
+
+
+class WithBackend(AbstractContextManager, abc.ABC):
```

Comment:
Is it still a context manager? Will be `close` called?

Reply:
This will never instantiated directly. In this design the children implement their own `__exit__` methods, as it is in `Project` (just closing) and `NeptuneObject` (closing + a lot of other things)

Reply:
I know but what's the reason then to have it here?

Reply:
To save typing :smile: all of its children will be context managers, right? So we will not need to add `AbstractContextManager` to their superclasses

Reply:
🫠